### ISSUE TITLE
v0.8.10 errors with "Cannot read properties of null (reading 'localName')"

### ISSUE THREAD

**Issue by XhmikosR** [xmldom/xmldom#583]
**Describe the bug**

```
Cannot read properties of null (reading 'localName')
```

**Expected behavior**

No errors

**Runtime & Version:**

xmldom version: `> 0.8.8`, e.g **`0.8.10`**
runtime version: N/A
other related software and version: N/A

**Additional context**

We hit the issue in svg-sprite and thus in projects that depend on it. Example: https://github.com/twbs/icons/actions/runs/6935431138/job/18865536634#step:6:22

I had to pin the version to 0.8.8 in svg-sprite so that we don't get these errors. I didn't try v0.8.9, but I could give it a go later if it helps narrow down the problem.

**Comment by karfau** [comment]
It would be very helpful if you (or someone from svg-sprite) could provide a reproduction with
- an xml/svg snippet that causes the issue
- how exactly xmldom is used to reproduce the error

There usually is a link to a stackblitz template in the bug template which makes this very easy.

Thx

**Comment by XhmikosR** [comment]
v0.8.9 also works for what is worth. So, it's the change in v0.8.10 that is a breaking change for us.

Unfortunately, I don't have a StackBlitz and the stacktrace is misleading (because it points to the svg-sprite source code). I'll see if I manage to create a StackBlitz with a reproducible example later.

**Comment by XhmikosR** [comment]
Specifically, this change seems to be the offending one https://github.com/xmldom/xmldom/compare/0.8.9...0.8.10#diff-79db8c65cf49002eb2d5d7e81a17f3b73d94db6bb50c0c12f0c9f427aff4b11eR172

```diff
-return this[index] || null;
+return index >= 0 && index < this.length ? this[index] : null;
```

**Comment by karfau** [comment]
Any xml source and steps to be able to reproduce it are helpful. Could also be a dedicated small GitHub repo that i can clone to reproduce.
The more narrow the setup the better, but it's ok if it still involves svg-sprite

**Comment by karfau** [comment]
By the way, the change you point to, belongs to the only PR that is included in version 8.10.0:
https://github.com/xmldom/xmldom/pull/514
So most likely svg-sprite is trying to access an element after a DOM manipulation that no longer exists in the DOM.
Earlier versions didn't properly clean up in such cases, so that it was possible to access elements behind the index indicated by the updated `length` attribute.

**Comment by karfau** [comment]
since there has not been an update on this for nearly a year, I'm closing it until we have more information
